From 6340307d62b02631def949dccf03799b8f97fc99 Mon Sep 17 00:00:00 2001
From: Martine Lenders <m.lenders@fu-berlin.de>
Date: Fri, 23 Feb 2018 17:54:05 +0100
Subject: [PATCH] ccn-lite-riot: MTU target too big

The type of `NETOPT_MAX_PACKET_SIZE` is [defined as uint16_t](
http://bit.ly/2CfYPbi), but i->mtu is an `int` (which is 32-bit on many
platforms), so for some devices, that properly check this, CCN-lite
can't initialize.
---
 src/ccnl-riot/src/ccn-lite-riot.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/ccnl-riot/src/ccn-lite-riot.c b/src/ccnl-riot/src/ccn-lite-riot.c
index 0dc46c8..d14751d 100644
--- a/src/ccnl-riot/src/ccn-lite-riot.c
+++ b/src/ccnl-riot/src/ccn-lite-riot.c
@@ -203,11 +203,13 @@ ccnl_open_netif(kernel_pid_t if_pid, gnrc_nettype_t netreg_type)
     i->addr.sa.sa_family = AF_PACKET;
 
     int res;
-    res = gnrc_netapi_get(if_pid, NETOPT_MAX_PACKET_SIZE, 0, &(i->mtu), sizeof(i->mtu));
+    uint16_t mtu;
+    res = gnrc_netapi_get(if_pid, NETOPT_MAX_PACKET_SIZE, 0, &(mtu), sizeof(mtu));
     if (res < 0) {
         DEBUGMSG(ERROR, "error: unable to determine MTU for if=<%u>\n", (unsigned) i->if_pid);
         return -ECANCELED;
     }
+    i->mtu = (int)mtu;
     DEBUGMSG(DEBUG, "interface's MTU is set to %i\n", i->mtu);
 
     res = gnrc_netapi_get(if_pid, NETOPT_ADDR_LEN, 0, &(i->addr_len), sizeof(i->addr_len));
-- 
2.7.4

